diff --git a/cpp/nanobind/CMakeLists.txt b/cpp/nanobind/CMakeLists.txt
index 03cf0dc..3c9a90a 100644
--- a/cpp/nanobind/CMakeLists.txt
+++ b/cpp/nanobind/CMakeLists.txt
@@ -18,7 +18,7 @@ target_sources(python_methods PRIVATE python_methods.cc)
 target_link_libraries(python_methods PUBLIC xgrammar)
 
 # Any code that uses nanobind directly lives here
-nanobind_add_module(xgrammar_bindings LTO nanobind.cc)
+nanobind_add_module(xgrammar_bindings nanobind.cc)
 target_link_libraries(xgrammar_bindings PRIVATE python_methods)
 
 if(DEFINED SKBUILD_PROJECT_NAME)
